### ISSUE TITLE
[Bug] Maximum call stack size exceeded

### ISSUE THREAD

**Issue by alephpt** [xenova/transformers.js#153]
When I pass in an large set  of 'training data' as input I get `Maximum call stack size exceeded

Here's my code
```
// load all .txt files from ../assets/
const files = fs.readdirSync('./assets/').filter(file => file.endsWith('.txt'));
const data = [];

// open all of the files and read them into a single string
for (let i = 0; i < files.length; i++) {
    const file = files[i];
    const fileContents = fs.readFileSync(`./assets/${file}`, 'utf-8');
    data.push(fileContents);
}
```

Calling the Data  with
```
let response = await generator(text + prompt + "\n\nResponse: ",
        {
            max_new_tokens: 1200,
            temperature: 0.5,
            top_k: 50,
            top_p: 0.3,
            repetition_penalty: 2.3,
            num_return_sequences: 1
        }
    )
```

Here's the response from the logs
```
RangeError: Maximum call stack size exceeded
    at Function.post_process (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/tokenizers.js:1236:30)
    at Function._call (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/tokenizers.js:1162:21)
    at Function.closure [as post_processor] (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/utils/core.js:62:28)
    at Function.encode (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/tokenizers.js:2115:35)
    at file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/tokenizers.js:1946:45
    at Array.map (<anonymous>)
    at Function._call (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/tokenizers.js:1946:31)
    at Function.closure [as tokenizer] (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/utils/core.js:62:28)
    at Function._call (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/pipelines.js:400:30)
    at closure (file:///home/sister/repos/bots/glaza/node_modules/@xenova/transformers/src/utils/core.js:62:28)
```

Running
```
Linux 6.3.4
Node 20.2.0
```

I'm sure there could be an error in my code, implementation or logic.. but if this is _not_ wrong, I would be happy to help resolve the bug or any issues this may have exposed.

**Comment by xenova** [comment]
Fixed by https://github.com/xenova/transformers.js/commit/4d5ecb4e9290bf9eef912dce0ec2c8cd147059dd

I've also added a test case for extremely long strings: https://github.com/xenova/transformers.js/commit/cbf18777c14ae53d722b69e47c4dddd7e0647f20. Before these changes, this would fail. Now, it succeeds 👍

**Comment by xenova** [comment]
These changes are now available in v2.3.0: https://www.npmjs.com/package/@xenova/transformers/v/2.3.0 🥳
